### ISSUE TITLE
Add an event cluster detection method?

### ISSUE THREAD

**Issue by cdeil** [gammapy/gammapy#496]
The standard method for source detection in gamma-ray astronomy is via a likelihood ratio test, i.e. to 
use TS and residual TS images similar to what we already do in gammapy.detect (see [here](https://gammapy.readthedocs.org/en/latest/detect/index.html#computation-of-ts-maps)).

The procedure to produce source catalogs like the 1FGL, 2FGL, 3FGL or even high-energy catalogs like the 1FHL or 2FHL is more complex (see links to catalogs and papers at http://fermi.gsfc.nasa.gov/ssc/data/access/lat/ ). It starts with wavelet detection methods to derive seed catalogs and then does a global iterative source detection / localisation method with coupled ROIs.

But for low-counts, high-energy Fermi data people are also using event cluster detection methods, see e.g.:
- http://adsabs.harvard.edu/abs/2013Ap%26SS.347..169C
- http://adsabs.harvard.edu/abs/2015MNRAS.452.3159A
- http://adsabs.harvard.edu/abs/2010arXiv1011.0210N

Implementing one of those in `gammapy.detect` would be welcome.
If I understand correctly, some people just use `DBSCAN` directly, which is readily available as [sklearn.cluster.DBSCAN](http://scikit-learn.org/stable/modules/generated/sklearn.cluster.DBSCAN.html).
In that case, all that would be needed would be documentation and some science verification (and maybe a thin wrapper around DBSCAN that is aware of sky coordinates or something).

I think having one cluster detection method available and documented in `gammapy.detect` would be nice. Preferably one that doesn't require much code on our side, i.e. that's already available e.g. in scikit-learn, or one that can be implemented with just 10s of lines of code.

**Comment by woodmd** [comment]
Actually for the most recent LAT catalogs (e.g. the 3FGL and now the internal releases of the 4FGL) seed-finding is primarily done with the likelihood ratio method as evaluated with pointlike.  With respect to the clustering or wavelet methods, the likelihood ratio method is more sensitive and works better for confused regions (i.e. finding faint sources near bright ones).   The only real advantage of the other methods is that they can be applied when you don't have a model for the diffuse emission.  However in the case it's not clear how useful the seeds would be since you would generally still need a diffuse model to do hypothesis testing and extract source characteristics.

For the gammapy development I think the effort would be better spent expanding the capabilities of the existing likelihood ratio implementation.  In fermipy I've implemented a likelihood ratio source-finding tool which uses the core functions from `gammapy.detect.test_statistics` with some small modifications.  Perhaps we should think about merging those changes back into `gammapy.detect.test_statistics` with the goal that I would eventually switch to using the gammapy implementation in fermipy.

**Comment by cdeil** [comment]
@woodmd - Thanks for the comment!

I completely agree with your assessment that likelihood-based methods are much more important than the event cluster finding methods.
The reason I described this was as an optional fun project for the last week of @OlgaVorokh's GSoC (see here [proposal](https://github.com/astropy/astropy/wiki/GSoC-2016-Application-Olga-Vorokh:-Image-processing-and-source-detection-in-Gammapy)).
And there are good use cases for it, like you say if no good diffuse model is available.

**Comment by cdeil** [comment]
Closing this old issue now. Such event cluster finding methods would still be welcome in gammapy.detect, but our focus is now on binned map likelihood analysis.
